src/lsdb/dask/join_catalog_data.py

```diff
@@ -121,6 +122,7 @@ def join_catalog_data_on(
 
     partition_map = get_partition_map_from_alignment_pixels(join_pixels)
     meta_df = generate_meta_df_for_joined_tables([left, right], suffixes)
-    ddf = dd.from_delayed(joined_partitions, meta=meta_df)
+    divisions = get_pixels_divisions(list(partition_map.keys()))
```

Comment:
This probably works, but technically a dictionary's keys aren't ordered, so the divisions might not be in the right order. Could you instead use the join_pixels to get them?

Reply:
We currently sort the divisions in case the pixels are unordered.